client/blocks/support-article-dialog/dialog.jsx

```diff
@@ -1,55 +1,63 @@
 import getInlineSupportArticleActionIsExternal from 'calypso/state/selectors/get-inline-support-article-action-is-external';
 import getInlineSupportArticleActionLabel from 'calypso/state/selectors/get-inline-support-article-action-label';
 import getInlineSupportArticleActionUrl from 'calypso/state/selectors/get-inline-support-article-action-url';
 import getInlineSupportArticleBlogId from 'calypso/state/selectors/get-inline-support-article-blog-id';
 import getInlineSupportArticlePostId from 'calypso/state/selectors/get-inline-support-article-post-id';
-import {
-	getSupportArticleAlternatesForLocale,
-	isRequestingSupportArticleAlternates,
-	shouldRequestSupportArticleAlternates,
-} from 'calypso/state/support-articles-alternates/selectors';
 import SupportArticleHeader from './header';
 import Placeholders from './placeholders';
 
 import './style.scss';
 import './content.scss';
 
 const noop = () => {};
+const getPostKey = ( blogId, postId ) => ( { blogId, postId } );
 
-export const SupportArticleDialog = ( {
-	actionIsExternal,
-	actionLabel,
-	actionUrl,
-	closeSupportArticleDialog,
-	post,
-	postId,
-	postKey,
-	shouldRequestAlternates,
-	isRequestingAlternates,
-} ) => {
+const useSupportArticleAlternatePostKey = ( blogId, postId ) => {
+	const supportArticleAlternates = useSupportArticleAlternatesQuery( blogId, postId );
+	if ( supportArticleAlternates.isLoading ) {
+		return null;
+	}
+
+	if ( ! supportArticleAlternates.data ) {
+		return getPostKey( blogId, postId );
+	}
+
+	return getPostKey( supportArticleAlternates.data.blog_id, supportArticleAlternates.data.page_id );
+};
+
+export const SupportArticleDialog = () => {
 	const translate = useTranslate();
-	const isLoading = ! post || isRequestingAlternates;
-	const siteId = post?.site_ID;
-	const shouldQueryReaderPost = ! post && ! shouldRequestAlternates && ! isRequestingAlternates;
 	const { value: supportArticleId, closeModal } = useRouteModal( 'support-article' );
+	const actionUrl = useSelector( getInlineSupportArticleActionUrl );
+	const actionLabel = useSelector( getInlineSupportArticleActionLabel );
+	const actionIsExternal = useSelector( getInlineSupportArticleActionIsExternal );
 	const articleUrl = actionUrl ? actionUrl : 'https://support.wordpress.com?p=' + supportArticleId;
+	const requestBlogId = useSelector( getInlineSupportArticleBlogId );
+	const blogId = requestBlogId ?? SUPPORT_BLOG_ID;
+	const currentQueryArgs = useSelector( getCurrentQueryArguments );
+	let postId = useSelector( getInlineSupportArticlePostId );
+	if ( ! postId ) {
+		postId = parseInt( currentQueryArgs?.[ 'support-article' ], 10 );
+	}
+	const supportArticleAlternates = useSupportArticleAlternatesQuery( blogId, postId );
+	const postKey = useSupportArticleAlternatePostKey( blogId, postId );
+	const post = useSelector( ( state ) => getPostByKey( state, postKey ) );
+	const isLoading = ! post || supportArticleAlternates.isLoading;
```

Comment:
The `supportArticleAlternates.isLoading` conditions can be replaced with `! postKey`

Reply:
Good point, done in 1d54916d04a2226f290f6afe6f062e521ccb10a9.